1-mandatory/1-writers.js

```diff
@@ -50,31 +50,55 @@ Exercise 1:
   and insert the corresponding values to the place holders that are indicated in curly braces:
   "Hi, my name is {firstName} {lastName}. I am {age} years old, and work as a {occupation}."
 */
+// function logAllWriters() {
+//   // write your code to log all writers here
+// }
+
 function logAllWriters() {
-  // write your code to log all writers here
+  for (let i = 0; i < writers.length; i++) {
+    const writer = writers[i];
+    console.log(`Hi, my name is ${writer.firstName} ${writer.lastName}. I am ${writer.age} years old, and work as a ${writer.occupation}.`);
+  }
 }
 
+
 /*
 Exercise 2:
   Only `console.log()` out the writers who are in their 40s (meaning between 40 and 49)
   and not alive anymore. Use the below sentence format:
   "Writer {firstName} {lastName} died at {age} years old."
 */
 
+
+
 function logDeadWritersInTheirForties() {
-  // write your code here
+  for (let i = 0; i < writers.length; i++) {
+    const writer = writers[i];
+    if (writer.age >= 40 && writer.age <= 49 && !writer.alive) {
+      console.log(`Writer ${writer.firstName} ${writer.lastName} died at ${writer.age} years old.`);
+    }
+  }
 }
 
```

Comment:
The for loop works great, try to use  writers.forEach might make the code simpler.

Thanks Jan